librosa/beat.py

```diff
@@ -167,9 +212,16 @@ def beat_track(
             y=y, sr=sr, hop_length=hop_length, aggregate=np.median
         )
 
+    if sparse and onset_envelope.ndim != 1:
+        raise ParameterError("input array must be one-dimensional if sparse=True")
```

Comment:
`"sparse=True (default) does not support {}-dimensional inputs. Consider passing sparse=False or converting to mono.`

with `{}` formatting to `onset_envelope.ndim`